src/data/viteTrickFetcher.ts

```diff
@@ -0,0 +1,158 @@
+/**
+ * **WARNING** This module is expected to be invoked from a Vite Plugin!
+ * 
+ * Aggregated all YAML Files in the ./tricks Directory and creates a list 
+ */
+
+
+// unfortunately, vite features (e.g. import.meta.glob) are NOT available to plugin code.
+// as such, we must fall back to reading YAMLs manually.
+import {globby} from "globby"
+import {join} from "path";
+import {fileURLToPath} from "url"
+import { ZodError, z } from "zod";
+import { DbTricksTableZod } from "../lib/database/schemas/CurrentVersionSchema"
+import {parse} from "yaml"
+import { readFile } from "fs/promises";
+import { YamlTrickTableSchemaZod } from "./viteSchemaGenerator";
+import buildGraph from "./variationOfCycleChecker";
+
+async function fetchYamlFile(path: string): Promise<z.infer<typeof DbTricksTableZod>> {
```

Comment:
you could omit the return type here as well

Reply:
done